lib/Artwork.php

```diff
@@ -681,10 +705,29 @@ public function Save(): void{
 			where
 			ArtworkId = ?
 		', [$this->Artist->ArtistId, $this->Name, $this->UrlName, $this->CompletedYear, $this->CompletedYearIsCirca,
-				$this->Created, $this->Status, $this->SubmitterUserId, $this->ReviewerUserId, $this->MuseumUrl, $this->PublicationYear, $this->PublicationYearPageUrl,
+				$this->Created, $this->Updated, $this->Status, $this->SubmitterUserId, $this->ReviewerUserId, $this->MuseumUrl, $this->PublicationYear, $this->PublicationYearPageUrl,
 				$this->CopyrightPageUrl, $this->ArtworkPageUrl, $this->IsPublishedInUs, $this->EbookWwwFilesystemPath, $this->MimeType, $this->Exception, $this->Notes,
 				$this->ArtworkId]
 		);
+
+		Db::Query('
+			DELETE FROM ArtworkTags
+			WHERE
+			ArtworkId = ?
+		', [$this->ArtworkId]
+		);
+
+		foreach($this->Tags as $tag){
+			Db::Query('
+				INSERT INTO ArtworkTags (ArtworkId, TagId)
+				VALUES (?,
+				        ?)
+			', [$this->ArtworkId, $tag->TagId]);
+		}
+
+		if(!empty($uploadedFile) && $uploadedFile['error'] == UPLOAD_ERR_OK){
```

Comment:
By the time we get here, we should already know that the upload was successful. If it wasn't then it should have thrown an exception in `Validate()`. So no need to check for error status here.

Reply:
Since the upload is optional during `Save()` these are the checks necessary to see if the user uploaded a valid file during their edit. I agree it looks like validation logic, so I added a comment here. I believe the check is necessary. I can elaborate more if the comment still isn't clear.